llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1312,7 +1312,7 @@ bool SPIRVToLLVM::postProcessRowMajorMatrix() {
               Value *pointerElem = getBuilder()->CreateGEP(pointerElemType, pointer, indices);
               Type *castType = GetElementPtrInst::getIndexedType(pointerElemType, indices);
               // TODO: Remove this when LLPC will switch fully to opaque pointers.
-              assert(castType == pointerElem->getType()->getNonOpaquePointerElementType());
+              assert(cast<PointerType>(pointerElem->getType())->isOpaqueOrPointeeTypeMatches(castType));
```

Comment:
I would prefer to have this a bit cleaned up, this is a lot of stuff to read. Can you use a macro or some other compile-time construct to make that a bit more readable?